SNMP Plugin - Telegraf crash when mib file is missing

### Relevant telegraf.conf

```toml
[agent]
  snmp_translator = "gosmi"

[[inputs.snmp]]
  agents = ["udp://xxx.xxx.xxx.xxx:161"]

  timeout = "15s"
  version = 2
  community = "CCCCCCCCCCC"
  retries = 10
  max_repetitions = 10
  #interval = "5m"
  interval = "24h"
  unconnected_udp_socket = true
  agent_host_tag = "source"
  tagexclude = ["host"]

  [[inputs.snmp.field]]
    # "An administratively-assigned name for this managed\n node. By convention, this is the node\'s fully-qualified\n domain name. If the name is unknown, the value is\n the zero-length string."
    oid = ".1.3.6.1.2.1.1.5.0"
    name = "SNMPv2-MIB::sysName"
    is_tag = true

  [[inputs.snmp.table]]
    # "hwIFExtIndex is the unique identifier of the table. It is written when the table is created and cannot be modified later.\n The table describes the extended attributes of an interface, such as the Layer 2/Layer 3 identifier and frame type of the interface."
    #oid = ".1.3.6.1.4.1.2011.5.25.41.1.1.1"
    name = "HUAWEI-IF-EXT-MIB::hwIFExtTable"
    inherit_tags = ["SNMPv2-MIB::sysName"]
    index_as_tag = true


  [[inputs.snmp.table.field]]
    # "This object indicates the interface index."
    oid = ".1.3.6.1.4.1.2011.5.25.41.1.1.1.1.1"
    name = "HUAWEI-IF-EXT-MIB::hwIFExtIndex"

  [[inputs.snmp.table.field]]
    # "This object indicates the Layer 2/Layer 3 switching flag of an interface.\n According to the switching command, note the following:\n When Layer2(1) is 1, the interface switches from Layer 3 mode to Layer 2 mode. When Layer3(2) is 2, the interface switches from Layer 2 mode to Layer 3 mode."
    oid = ".1.3.6.1.4.1.2011.5.25.41.1.1.1.1.2"
    name = "HUAWEI-IF-EXT-MIB::hwIFExtLayer"
    conversion = "enum(1)"

  [[inputs.snmp.table.field]]
    # "This object indicates the type of frames that a VLAN virtual interface receives. \n Frame formats include ethernetII(1), ethernetII(2), ethernet8022(3), and ethernet8023. \n The value for the ethernetII(1) format is 1. Currently, only ethernetII(1) is supported."
    oid = ".1.3.6.1.4.1.2011.5.25.41.1.1.1.1.3"
    name = "HUAWEI-IF-EXT-MIB::hwIFExtFrameType"
    conversion = "enum(1)"

  [[inputs.snmp.table.field]]
    # "This object indicates the interval for collecting traffic statistics for an interface. The value ranges from 0 to 600, in seconds. The default value is 300."
    oid = ".1.3.6.1.4.1.2011.5.25.41.1.1.1.1.4"
    name = "HUAWEI-IF-EXT-MIB::hwIFExtFlowStatInterval"

```


### Logs from Telegraf

```text
There is no relevant log
```


### System info

Telegraf version: 1.32.0 OS: SUSE

### Docker

_No response_

### Steps to reproduce

When running telegraf with snmp pluging, if a needed mib file is missing following error occurs.
The solution is simple, putting the missing mib file into /usr/share/snmp/mibs directory or the configured one.

### Expected behavior

Trap this error and log the mib file that is missing in order to make easy to locate the problem.

### Actual behavior

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xcc39ed]

goroutine 1040 [running]:
github.com/sleepinggenius2/gosmi/models.Node.FormatValue(...)
        /go/pkg/mod/github.com/sleepinggenius2/gosmi@v0.4.4/models/format.go:127
github.com/influxdata/telegraf/internal/snmp.(*gosmiTranslator).SnmpFormatEnum(0xc002011ac1?, {0xc00406bbc0?, 0x9f7c5c8?}, {0x7be8e00, 0xf9726a8}, 0x1)
        /go/src/github.com/influxdata/telegraf/internal/snmp/translator_gosmi.go:69 +0x18d
github.com/influxdata/telegraf/internal/snmp.(*Field).Convert(0xc0062f77a0, {{0x7be8e00, 0xf9726a8}, {0xc00406bbc0, 0x28}, 0x2})
        /go/src/github.com/influxdata/telegraf/internal/snmp/field.go:302 +0xf47
github.com/influxdata/telegraf/internal/snmp.Table.Build.func1({{0x7be8e00, 0xf9726a8}, {0xc00406bbc0, 0x28}, 0x2})
        /go/src/github.com/influxdata/telegraf/internal/snmp/table.go:208 +0x268
github.com/gosnmp/gosnmp.(*GoSNMP).walk(0xc005f9cb00, 0xa5, {0xc002014451?, 0xc0066ecd50?}, 0xc0066ecd50)
        /go/pkg/mod/github.com/gosnmp/gosnmp@v1.37.0/walk.go:152 +0xcda
github.com/gosnmp/gosnmp.(*GoSNMP).BulkWalk(...)
        /go/pkg/mod/github.com/gosnmp/gosnmp@v1.37.0/gosnmp.go:590
github.com/influxdata/telegraf/internal/snmp.GosnmpWrapper.Walk({0xc00649f980?}, {0xc002014451?, 0xc00649f980?}, 0x13?)
        /go/src/github.com/influxdata/telegraf/internal/snmp/wrapper.go:40 +0x45
github.com/influxdata/telegraf/internal/snmp.Table.Build({{0xc0020143c1, 0x22}, {0xc002f42e50, 0x1, 0x1}, 0x1, {0xc0001ae008, 0xc, 0xc}, {0x0, ...}, ...}, ...)
        /go/src/github.com/influxdata/telegraf/internal/snmp/table.go:182 +0x4b1
github.com/influxdata/telegraf/plugins/inputs/snmp.(*Snmp).gatherTable(0xc001fb2780, {0xa07b000, 0xc006743300}, {0xa0371c0, 0xc005f9cb00}, {{0xc0020143c1, 0x22}, {0xc002f42e50, 0x1, 0x1}, ...}, ...)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:134 +0x87
github.com/influxdata/telegraf/plugins/inputs/snmp.(*Snmp).Gather.func1(0x0?, {0xc0027a6701, 0x16})
        /go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:122 +0x448
created by github.com/influxdata/telegraf/plugins/inputs/snmp.(*Snmp).Gather in goroutine 136
        /go/src/github.com/influxdata/telegraf/plugins/inputs/snmp/snmp.go:102 +0x66


### Additional info

_No response_

--- a/internal/snmp/translator_gosmi.go
+++ b/internal/snmp/translator_gosmi.go
@@ -64,6 +64,10 @@ func (g *gosmiTranslator) SnmpFormatEnum(oid string, value interface{}, full boo
 		return "", err
 	}
 
+	if node.Type == nil {
+		return "", fmt.Errorf("cannot format %v as enum, unkown OID", oid)
+	}
+
 	var v models.Value
 	if full {
 		v = node.FormatValue(value, models.FormatEnumName, models.FormatEnumValue)
@@ -71,7 +75,7 @@ func (g *gosmiTranslator) SnmpFormatEnum(oid string, value interface{}, full boo
 		v = node.FormatValue(value, models.FormatEnumName)
 	}
 
-	return v.Formatted, nil
+	return v.String(), nil
 }
 
 func (g *gosmiTranslator) SnmpFormatDisplayHint(oid string, value interface{}) (string, error) {
@@ -85,9 +89,11 @@ func (g *gosmiTranslator) SnmpFormatDisplayHint(oid string, value interface{}) (
 		return "", err
 	}
 
-	v := node.FormatValue(value)
+	if node.Type == nil {
+		return "", fmt.Errorf("cannot format %v as displayhint, unkown OID", oid)
+	}
 
-	return v.Formatted, nil
+	return node.FormatValue(value).String(), nil
 }
 
 func getIndex(mibPrefix string, node gosmi.SmiNode) (col []string, tagOids map[string]struct{}) {

--- a/internal/snmp/translator_gosmi.go
+++ b/internal/snmp/translator_gosmi.go
@@ -12,6 +12,8 @@ import (
 	"github.com/influxdata/telegraf"
 )
 
+var errCannotFormatUnkownType = errors.New("cannot format value, unknown type")
+
 type gosmiTranslator struct {
 }
 
@@ -65,7 +67,7 @@ func (g *gosmiTranslator) SnmpFormatEnum(oid string, value interface{}, full boo
 	}
 
 	if node.Type == nil {
-		return "", fmt.Errorf("cannot format %v as enum, unkown OID", oid)
+		return "", errCannotFormatUnkownType
 	}
 
 	var v models.Value
@@ -90,7 +92,7 @@ func (g *gosmiTranslator) SnmpFormatDisplayHint(oid string, value interface{}) (
 	}
 
 	if node.Type == nil {
-		return "", fmt.Errorf("cannot format %v as displayhint, unkown OID", oid)
+		return "", errCannotFormatUnkownType
 	}
 
 	return node.FormatValue(value).String(), nil

